### PR TITLE
Updated for view editor compatibility

### PR DESCRIPTION
**Pull request by BA-MBSE**
* Added search pagination and consistent ordering
* Added /checkAuth endpoint
* Added field wildcard

Updated ElasticSearchService to intelligently choose query methods based on field mapping types

### CODE REVIEW

File: authenticator/src/main/java/org/openmbee/sdvc/authenticator/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -47,4 +50,23 @@ public JwtAuthenticationResponse createAuthenticationToken(
 
     }
 
+    @GetMapping(value = "/checkAuth")
+    @PreAuthorize("isAuthenticated()")
+    public JwtTokenValidationResponse checkAuthenticationToken() {
+        final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication != null && authentication.getPrincipal() instanceof String) {
+            final String username = (String) authentication.getPrincipal();
+            if (username != null && username != "anonymousUser") {
```
Comment (HuiJun) [inline authenticator/src/main/java/org/openmbee/sdvc/authenticator/controllers/AuthenticationController.java]: Is "anonymousUser" a standard user that can be created by some dependency? Or is it a user that will be required?

Comment (dlamoris) [reply authenticator/src/main/java/org/openmbee/sdvc/authenticator/controllers/AuthenticationController.java]: i think that's what spring puts in when a user isn't authenticated (authenticated anonymously) 
maybe a better way would be to just check for Authentication object instance of AnonymousAuthenticationToken, 
and if it's not, just do authentication.getName() (getName already checks for userdetails/principal and calls the right things)

**Review by dlamoris — CHANGES_REQUESTED** [review]
the check auth can be simplified (see MethodSecurityService), otherwise looks good

File: authenticator/src/main/java/org/openmbee/sdvc/authenticator/controllers/AuthenticationController.java (hunk 1 above)
Comment (BA-MBSE) [reply authenticator/src/main/java/org/openmbee/sdvc/authenticator/controllers/AuthenticationController.java]: Simplified as requested

**Review by dlamoris — APPROVED** [review]

**Pull request merged**